posthog/temporal/data_imports/pipelines/pipeline/pipeline.py

```diff
@@ -250,6 +259,32 @@ def _process_pa_table(self, pa_table: pa.Table, index: int):
 
         _update_job_row_count(self._job.id, pa_table.num_rows, self._logger)
 
+        # if first sync, we make the delta table files available for querying and create the data warehouse table, so
+        # that the user has some data available to start using
+        if is_first_sync:
+            self._logger.debug(
+                "First sync, making delta table files available for querying and creating data warehouse table"
+            )
+            file_uris = delta_table.file_uris()
+            new_file_uris = [uri for uri in file_uris if uri not in previous_file_uris]
+            self._logger.debug(f"Adding {len(new_file_uris)} S3 files to query folder")
+            prepare_s3_files_for_querying(
+                folder_path=self._job.folder_path(),
+                table_name=self._resource_name,
+                file_uris=new_file_uris,
+                # delete existing files if it's the first chunk, otherwise we'll just append to the existing files
+                delete_existing=False if index > 0 else True,
+            )
+            self._logger.debug("Validating schema and updating table")
+            validate_schema_and_update_table_sync(
```

Comment:
This does make a call to clickhouse by the way - it should be fine, we're using a local clickhouse server on the pod and not our cluster, but still something to be aware of